master/cmd/determined-master/root.go

```diff
@@ -73,6 +73,18 @@ func initializeConfig() error {
 	if err != nil {
 		return err
 	}
+
+	// Validate warnings for extraneous and duplicate keys before
+	// Viper merges configuration. Viper does some weird merging logic
+	// that appears to dedupe some values along with ignoring config like
+	// not_a_real_key: {}
+	var c *config.Config
+	if err = yaml.UnmarshalStrict(bs, &c, yaml.DisallowUnknownFields); err != nil {
+		log.Warnf("master configuration parsing warning (will soon be error) %s", err)
```

Comment:
nit: wording feels weird here - are you saying that it will become an error in the future? Or that it will error later in the code? If the former, I'd just call it a warning and `TODO` the comment change, the latter I'd just call it an error.